rust/automerge-c/src/result.rs

```diff
@@ -749,17 +737,17 @@ pub enum AMstatus {
 /// \brief Gets a result's error message string.
 ///
 /// \param[in] result A pointer to an `AMresult` struct.
-/// \return A UTF-8 string value or `NULL`.
+/// \return A UTF-8 string view as an `AMbyteSpan` struct.
 /// \pre \p result `!= NULL`.
 /// \internal
 ///
 /// # Safety
 /// result must be a valid pointer to an AMresult
 #[no_mangle]
-pub unsafe extern "C" fn AMerrorMessage(result: *const AMresult) -> *const c_char {
+pub unsafe extern "C" fn AMerrorMessage(result: *const AMresult) -> AMbyteSpan {
```

Comment:
@jkankiewicz is it possible to leave this as a `*const c_char` - as the authors of automerge we can probably ensure that error messages don't contain null bytes; and that does make error handling a bit more awkward for C users, for no real benefit to wrapper languages.